homeassistant/components/bluetooth/__init__.py

```diff
@@ -178,14 +181,83 @@ def async_track_unavailable(
     return manager.async_track_unavailable(callback, address)
 
 
+async def _async_has_bluetooth_adapter() -> bool:
+    """Return if the device has a bluetooth adapter."""
+    if platform.system() == "Darwin":  # CoreBluetooth is built in on MacOS hardware
+        return True
+    if platform.system() == "Windows":  # We don't have a good way to detect on windows
+        return False
+    return bool(await _async_get_bluetooth_adapters())
+
+
+async def _async_get_bluetooth_adapters() -> set[str]:
+    """Return a list of bluetooth adapters."""
+    adapters: set[str] = set()
```

Comment:
Should probably do late imports for `dbus_next` since there is no point in loading it on windows/mac